src/main/java/org/jabref/gui/openoffice/StyleSelectDialogView.java

```diff
@@ -163,40 +203,95 @@ private void initialize() {
 
         add.setGraphic(IconTheme.JabRefIcons.ADD.getGraphicNode());
 
+        // JStyle previews
+        previewArticle = initializePreviewViewer(TestEntry.getTestEntry());
+        jstylePreviewBox.getChildren().add(previewArticle);
+
+        previewBook = initializePreviewViewer(TestEntry.getTestEntryBook());
+        jstylePreviewBox.getChildren().add(previewBook);
+
         EasyBind.subscribe(viewModel.selectedItemProperty(), style -> {
             if (viewModel.getSelectedStyle() instanceof JStyle) {
                 tvStyles.getSelectionModel().select(style);
                 previewArticle.setLayout(new TextBasedPreviewLayout(style.getJStyle().getReferenceFormat(StandardEntryType.Article)));
                 previewBook.setLayout(new TextBasedPreviewLayout(style.getJStyle().getReferenceFormat(StandardEntryType.Book)));
             }
         });
+    }
 
-        availableListView.setItems(viewModel.getAvailableLayouts());
-        searchBox.textProperty().addListener((observable, oldValue, newValue) ->
-                viewModel.setAvailableLayoutsFilter(newValue));
+    private void setupCslStylesTab() {
+        // Set up CSL styles table columns
+        cslNameColumn.setCellValueFactory(cellData -> cellData.getValue().nameProperty());
+        cslPathColumn.setCellValueFactory(cellData -> cellData.getValue().pathProperty());
+        cslDeleteColumn.setCellValueFactory(cellData -> cellData.getValue().internalStyleProperty());
 
-        viewModel.setSelectedTab(tabPane.getSelectionModel().getSelectedItem());
-        tabPane.getSelectionModel().selectedItemProperty().addListener((observable, oldValue, newValue) -> viewModel.setSelectedTab(newValue));
+        new ValueTableCellFactory<CitationStyleViewModel, Boolean>()
+                .withGraphic(internalStyle -> {
+                    if (!internalStyle) {
+                        return IconTheme.JabRefIcons.DELETE_ENTRY.getGraphicNode();
+                    }
+                    return null;
+                })
```

Comment:
what style do we prefer? if block or oneliner?
```
                .withGraphic(internalStyle -> (!internalStyle) ? IconTheme.JabRefIcons.DELETE_ENTRY.getGraphicNode() : null)
```

Reply:
for this I would prefer a one liner